ddtrace/debugging/_capture/utils.py

```diff
@@ -140,14 +142,24 @@ def capture_exc_info(exc_info):
     }
 
 
-def capture_value(value, level=MAXLEVEL, maxlen=MAXLEN, maxsize=MAXSIZE, maxfields=MAXFIELDS):
-    # type: (Any, int, int, int, int) -> Dict[str, Any]
+def capture_value(value, level=MAXLEVEL, maxlen=MAXLEN, maxsize=MAXSIZE, maxfields=MAXFIELDS, hg=None):
+    # type: (Any, int, int, int, int, Optional[HourGlass]) -> Dict[str, Any]
```

Comment:
I like the time_permits function - maybe we can simplify this interface and instead of Optional[HourGlass] have a cancellation callback that stops this loop. basically pass time_permits instead of hg.

The problem I have with this function is that cancellation callback not necessarily is a timeout. so that function might return string with the reason.

so, in snapshot it would be: capture_value(..., lambda () => return "timeout" if ! hg.trickling() else None)

WDYT?

Reply:
👍  I'll have a think about this.